src/v/kafka/requests/init_producer_id_request.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "bytes/iobuf.h"
+#include "kafka/errors.h"
+#include "kafka/requests/request_context.h"
+#include "kafka/requests/response.h"
+#include "kafka/requests/schemata/init_producer_id_request.h"
+#include "kafka/requests/schemata/init_producer_id_response.h"
+#include "kafka/types.h"
+#include "model/fundamental.h"
+#include "model/timestamp.h"
+#include "seastarx.h"
+
+#include <seastar/core/future.hh>
+
+#define SUPPRESS_IDEMPOTENT_PRODUCER
```

Comment:
interesting. how come we need a macro. 

if we don't use the api, it should be fine still no ? LTO will eliminate the code, altogether.

Reply:
Right now (before this PR or with the macro) when a client tries to use idempotent producers she gets an explicit error message like "idempotent producers are not supported in the current version" but if I remove the macro the connection gets terminated and she gets a generic error like "something went wrong". The first error message is way more explicit so I prefer to keep it until the feature is done.

Reply:
I think you can remove the pre-processor guard and return from the request proceessor:

```
136     return ss::make_exception_future<response_ptr>(¬                                                                                                                                                                                                                                 
137       std::runtime_error(fmt::format("Unsupported API {}", ctx.header().key)));¬          
```

to get the same affect without the macros.

Reply:
or maybe the client is making the nice error message from the results of the api versions request, in which case you could _not_ register the request type with the api versions list.

Reply:
oh i see you tried the runtime_error thing.

Reply:
yes, you're right, the error message depends in the api versions response, I'll  exclude the init_producer_id from the request_types list and get rid or the macro

Reply:
removed macro and didn't include init_producer_id to api_versions_request